src/platform/backends/hyperv_api/hcs/hyperv_hcs_compute_system_state.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#pragma once
+
+#include <algorithm>
+#include <cstdint>
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+#include <fmt/format.h>
+
+namespace multipass::hyperv::hcs
+{
+
+/**
+ * Enum values representing a compute system's possible state
+ *
+ * @ref https://learn.microsoft.com/en-us/virtualization/api/hcs/schemareference#State
+ */
+enum class ComputeSystemState : std::uint8_t
+{
+    created,
+    running,
+    paused,
+    stopped,
+    saved_as_template,
+    unknown,
+};
+
+/**
+ * Translate host compute system state string to enum
+ *
+ * @param str
+ * @return ComputeSystemState
+ */
+[[nodiscard]] inline std::optional<ComputeSystemState> compute_system_state_from_string(
+    std::string str)
+{
+    std::transform(str.begin(), str.end(), str.begin(), [](unsigned char c) {
+        return std::tolower(c);
+    });
+
+    static const std::unordered_map<std::string, ComputeSystemState> translation_map{
```

Comment:
I am not asking for that now, but you could have defined this as an array of pairs outside of this function and then just iterated either way (here and below).

Reply:
I can implement a simple bimap type with two unordered_map's for now. How does that sound to you?

Reply:
Done